packages/peregrine/lib/util/simplePersistence.js

```diff
@@ -42,16 +42,12 @@ export default class BrowserPersistence {
         return this.storage.getItem(name);
     }
     getItem(name) {
-        const now = Date.now();
         const item = this.storage.getItem(name);
         if (!item) {
             return undefined;
         }
-        const { value, ttl, timeStored } = JSON.parse(item);
-        if (ttl && now - timeStored > ttl * 1000) {
-            this.storage.removeItem(name);
-            return undefined;
-        }
+        const { value } = JSON.parse(item);
```

Comment:
Instead of doing a side effect in the getter function, I moved it into a `useEffect` in the `UserContextProvider` which seemed the right spot for it.

Reply:
What about other usages of this side effect like https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/Router/resolveUnknownRoute.js#L151 or in the possible customizations?

Reply:
Good question @eug123. I don't think that they are related to side effects. The file you mentioned is not used in our code base. Looks like it's dead code.

Reply:
Ok, I just found setting item with ttl, have not checked if this actually used. 
But it is still possible to add ttl - `setItem(name, value, ttl)`, so you will need to implement expiration logic by yourself, whenever you need it?

Reply:
We do have it, its part of User Context file.